dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue

```diff
@@ -1,139 +1,452 @@
+import { modifyObjectKeys, objectType, updateObject } from "@/utils/UpdateObjectUtils";
+import { formatBytesUserFriendly } from "@/utils/NumberConversionUtils";
+import { ExtendedCompanyReport, ExtendedFile, WhichSetOfFiles } from "@/components/forms/Types";
+import JumpLinksSection from "@/components/forms/parts/JumpLinksSection.vue";
 
 export default defineComponent({
   name: "CreateEUTaxonomyForNonFinancials",
-  components: { DataPointFormElement, FailedUpload, Card, FormKit, SuccessUpload },
+  components: {
+    JumpLinksSection,
+    Calendar,
+    UploadFormHeader,
+    PrimeButton,
+    UploadReports,
+    BasicInformationFields,
+    KPIfieldSet,
+    FailedUpload,
+    Card,
+    FormKit,
+    SuccessUpload,
+  },
   setup() {
     return {
       getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
     };
   },
 
   data: () => ({
-    innerClass: {
-      "formkit-inner": false,
-      "p-inputwrapper": true,
-    },
-    inputClass: {
-      "formkit-input": false,
-      "p-inputtext": true,
-      "w-full": true,
+    formInputsModel: {} as CompanyAssociatedDataEuTaxonomyDataForNonFinancials,
+    fiscalYearEnd: "" as Date | "",
+    convertedFiscalYearEnd: "",
+    reportingPeriod: new Date(),
+    filesToUpload: [] as ExtendedFile[],
+    uploadFiles: [] as ExtendedCompanyReport[],
+    onThisPageLinks: [
+      { label: "Upload company reports", value: "uploadReports" },
+      { label: "Basic information", value: "basicInformation" },
+      { label: "Assurance", value: "assurance" },
+      { label: "CapEx", value: "capex" },
+      { label: "OpEx", value: "opex" },
+      { label: "Revenue", value: "revenue" },
+    ],
+    elementPosition: 0,
+    route: useRoute(),
+    editMode: false,
+    waitingForData: false,
+    checkCustomInputs,
+    formatBytesUserFriendly,
+    maxFileSize: UPLOAD_MAX_FILE_SIZE_IN_BYTES,
+    euTaxonomyKPIsModel: euTaxonomyPseudoModelAndMappings,
+    euTaxonomyKpiNameMappings,
+    euTaxonomyKpiInfoMappings,
+    reportingPeriodYear: new Date().getFullYear(),
+    assuranceData: {
+      None: humanizeString("None"),
+      LimitedAssurance: humanizeString("LimitedAssurance"),
+      ReasonableAssurance: humanizeString("ReasonableAssurance"),
     },
+
     postEuTaxonomyDataForNonFinancialsProcessed: false,
     messageCount: 0,
-    formInputsModel: {},
     postEuTaxonomyDataForNonFinancialsResponse: null,
     humanizeString: humanizeString,
   }),
+  watch: {
+    reportingPeriod: function (newValue: Date) {
+      this.reportingPeriodYear = newValue.getFullYear();
+    },
+  },
+  computed: {
+    namesOfFilesToUpload(): string[] {
+      const namesFromFilesToUpload = this.filesToUpload.map((el) => el.name.split(".")[0]);
+      const namesFromUploadedFiles = this.uploadFiles.map((el) => el.name.split(".")[0]);
+      return [...new Set([...namesFromFilesToUpload, ...namesFromUploadedFiles])];
+    },
+  },
```

Comment:
Also in this area please adpot any changes you make in `CreateEuTaxonomyFOrFinancials.vue`.